src/styles/calendar.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  width: ($day-size + $gap-size) * 7;
+  margin: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+
+  &__day {
+    width: #{$day-size};
+    height: #{$day-size};
+    border: #{$border-size} solid #{$border-color};
+    display: block;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: " #{$i}";
+      }
+    }
+
+    font-size: $number-font-size;
+    text-align: center;
+    line-height: $day-size;
+    transition: #{$transition-property} #{$transition-duration};
```

Comment:
You use the wrong syntax a bit, that's why your transitions don't work.
https://www.loom.com/share/b8d2283f59a544cc92ef58cdcab040f4?sid=bffca49e-afe3-4813-b722-c23def16162c

Also, `cursor: pointer;` you should apply to an element instead of just `:hover`